pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -252,39 +267,43 @@ func (cgc *containerGC) evictSandboxes(minAge time.Duration) error {
 		return err
 	}
 
-	evictSandboxes := make([]string, 0)
-	newestGCTime := time.Now().Add(-minAge)
+	evictSandboxes := make(sandboxesByPodUID)
 	for _, sandbox := range sandboxes {
```

Comment:
The new logic only keeps the latest sandbox for each pod, all other sandboxes (including running ones) should be removed.